pkg/processor/coordinator_helper.go

```diff
@@ -218,103 +236,129 @@ func (c *ConstructorHelper) Sign(
-			accountIdentifier.Address,
-		)
-	}
-
-	return val, err
+	return amount, err
 }
 
-// CoinBalance returns the balance of the largest
-// Coin owned by an address.
-func (c *ConstructorHelper) CoinBalance(
+// Coins returns all *types.Coin owned by
+// an account.
+func (c *CoordinatorHelper) Coins(
 	ctx context.Context,
+	dbTx storage.DatabaseTransaction,
 	accountIdentifier *types.AccountIdentifier,
 	currency *types.Currency,
-) (*big.Int, *types.CoinIdentifier, error) {
-	coinValue, coinIdentifier, _, err := c.coinStorage.GetLargestCoin(
+) ([]*types.Coin, error) {
+	coins, _, err := c.coinStorage.GetCoinsTransactional(
 		ctx,
+		dbTx,
 		accountIdentifier,
-		currency,
 	)
+	if err != nil {
+		return nil, fmt.Errorf("%w: unable to get coins", err)
+	}
+
+	coinsToReturn := []*types.Coin{}
+	for _, coin := range coins {
+		if types.Hash(coin.Amount.Currency) != types.Hash(currency) {
+			continue
+		}
 
-	return coinValue, coinIdentifier, err
+		coinsToReturn = append(coinsToReturn, coin)
+	}
+
+	return coinsToReturn, nil
 }
 
 // LockedAddresses returns a slice of all addresses currently sending or receiving
 // funds.
-func (c *ConstructorHelper) LockedAddresses(ctx context.Context) ([]string, error) {
-	return c.broadcastStorage.LockedAddresses(ctx)
+func (c *CoordinatorHelper) LockedAddresses(
+	ctx context.Context,
+	dbTx storage.DatabaseTransaction,
+) ([]string, error) {
+	return c.broadcastStorage.LockedAddresses(ctx, dbTx)
 }
 
 // AllBroadcasts returns a slice of all in-progress broadcasts in BroadcastStorage.
-func (c *ConstructorHelper) AllBroadcasts(ctx context.Context) ([]*storage.Broadcast, error) {
+func (c *CoordinatorHelper) AllBroadcasts(ctx context.Context) ([]*storage.Broadcast, error) {
 	return c.broadcastStorage.GetAllBroadcasts(ctx)
 }
 
 // ClearBroadcasts deletes all pending broadcasts.
-func (c *ConstructorHelper) ClearBroadcasts(ctx context.Context) ([]*storage.Broadcast, error) {
+func (c *CoordinatorHelper) ClearBroadcasts(ctx context.Context) ([]*storage.Broadcast, error) {
 	return c.broadcastStorage.ClearBroadcasts(ctx)
 }
 
 // Broadcast enqueues a particular intent for broadcast.
-func (c *ConstructorHelper) Broadcast(
+func (c *CoordinatorHelper) Broadcast(
 	ctx context.Context,
-	sender string,
+	dbTx storage.DatabaseTransaction,
+	identifier string,
+	network *types.NetworkIdentifier,
 	intent []*types.Operation,
 	transactionIdentifier *types.TransactionIdentifier,
 	payload string,
+	confirmationDepth int64,
 ) error {
 	return c.broadcastStorage.Broadcast(
 		ctx,
-		sender,
+		dbTx,
+		identifier,
+		network,
 		intent,
 		transactionIdentifier,
 		payload,
+		confirmationDepth,
 	)
 }
 
-// AllAddresses returns a slice of all known addresses.
-func (c *ConstructorHelper) AllAddresses(ctx context.Context) ([]string, error) {
-	return c.keyStorage.GetAllAddresses(ctx)
+// BroadcastAll attempts to broadcast all ready transactions.
+func (c *CoordinatorHelper) BroadcastAll(
```

Comment:
How does this method handle partial failures, i.e. if one broadcast fails, does the subsequent one also fails or it will continue broadcasting?

Reply:
The caller defines a `BroadcastLimit` for the max number of times to attempt re-broadcast (https://github.com/coinbase/rosetta-cli/blob/04f379541ad59a15a9cbf0e8472b12cedbcc2fa7/configuration/configuration.go#L186-L188) and if broadcast failures should be ignored (https://github.com/coinbase/rosetta-cli/blob/04f379541ad59a15a9cbf0e8472b12cedbcc2fa7/configuration/configuration.go#L190-L192).

TL;DR however you want!